templates/common/infra/bicep/core/monitor/applicationinsights.bicep

```diff
@@ -0,0 +1,32 @@
+param environmentName string
+param location string = resourceGroup().location
+param logAnalyticsWorkspaceId string
+
+var resourceToken = toLower(uniqueString(subscription().id, environmentName, location))
+var tags = { 'azd-env-name': environmentName }
+var abbrs = loadJsonContent('../../abbreviations.json')
+
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' = {
+  name: '${abbrs.insightsComponents}${resourceToken}'
+  location: location
+  tags: tags
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+    WorkspaceResourceId: logAnalyticsWorkspaceId
+  }
+}
+
+module applicationInsightsDashboard 'applicationinsights-dashboard.bicep' = {
+  name: 'application-insights-dashboard'
+  params: {
+    environmentName: environmentName
+    location: location
```

Comment:
consider taking the name of the app insights resource as a parameter here - it will let you remove the explicit depends on and then the module can be used for any app insights resource, not just ones that follow our naming convention?

Reply:
Done